src/hooks/useForm.js

```diff
@@ -0,0 +1,11 @@
+import { useState } from 'react';
+
+export const useForm = (initialState = {}) => {
+  const [values, setValues] = useState(initialState);
+
+  const handleInputChange = ({ target }) => {
+    setValues({ ...values, [target.name]: target.value });
```

Comment:
Really great hook and `handleInputChange` is so simple and elegant. :-)

There's one small thing that can lead to nasty and really hard bugs - state setting is asynchronous, meaning that you can't always trust that state will be the same between the time that you started setting it to when you finished setting it. If there is a ton of different state setting going on, then there exists the possibility of "race conditions." 

So, for this reason, it's usually best to avoid referencing the soon-to-be previous state that you're setting from within the setter. (sorry for saying "set" so much!) 

Now, in this case, `values` is the soon-to-be previous state and we are referencing it in `setValues` by spreading it in. It's a pretty low chance that this could cause a bug, but I am always really cautious referencing current state in my setter.   

Luckily, the people who make React are super smart and are fully aware of this issue. For this reason, they provide a callback in the setter, which passes the previous state as an argument and this previous state will always dependably be the same throughout the setting. What you return from this callback will get set as the new state. So, in this case, you could write this as something like the following:

```javascript
setValues(previousValues => ({...previousValues, [target.name]: target.value});
```
(note that this uses an implicit return. let me know if that syntax is new to you)

https://es.wikipedia.org/wiki/Condici%C3%B3n_de_carrera

Reply:
It is new to me 😅 and I am a little confused.. I am going to investigate about this